src/Core/Checkout/Cart/SalesChannel/CartOrderRoute.php

```diff
@@ -169,4 +167,23 @@ private function addAffiliateTracking(Cart $cart, DataBag $data): void
             $cart->setCampaignCode($campaignCode);
         }
     }
+
+    private function place(Cart $cart, SalesChannelContext $context, RequestDataBag $data): OrderPlaceResult
+    {
+        $calculatedCart = $this->cartCalculator->calculate($cart, $context);
+
+        $response = $this->checkoutGatewayRoute->load(new Request($data->all(), $data->all()), $cart, $context);
+        $calculatedCart->addErrors(...$response->getErrors());
+
+        $this->taxProviderProcessor->process($calculatedCart, $context);
+
+        $this->addCustomerComment($calculatedCart, $data);
+        $this->addAffiliateTracking($calculatedCart, $data);
+
+        Profiler::trace('checkout-order::pre-payment', fn() => $this->paymentProcessor->validate($calculatedCart, $data, $context));
+
+        $orderId = Profiler::trace('checkout-order::order-persist', fn() => $this->orderPersister->persist($calculatedCart, $context));
+
+        return new OrderPlaceResult($orderId);
```

Comment:
what is the benefit of having a class as return value and not the order ID directly?

Reply:
The intention is a more future proof extension point. If you want to add more information in the future, you dont have to break it.

When you just return the order id string, each additional data would be a break ;)